cmd/microcloud/main_init.go

```diff
@@ -533,8 +533,8 @@ func (c *initConfig) validateSystems(s *service.Handler) (err error) {
 			return fmt.Errorf("OVN underlay IP %q is invalid", sys.OVNGeneveAddr)
 		}
 
-		if subnet.Contains(underlayIP) {
-			fmt.Printf("Warning: OVN underlay IP (%s) is shared with the Ceph cluster network (%s)\n", underlayIP.String(), subnet.String())
+		if sys.OVNGeneveIface == sys.MicroCephInternalNetworkIface {
```

Comment:
There are potentially four different interfaces that could collide
* OVN uplink interface
* OVN geneve interface
* Ceph public/internal interface

I have the feeling we should check if there is a overlap between any of them, not just only the geneve and Ceph cluster ifaces.

Reply:
I updated a bit the approach. What do you think?